internal/api/impl/v1/globalrolebinding/service.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package globalrolebinding
+
+import (
+	"fmt"
+
+	"github.com/perses/perses/internal/api/interface/v1/globalrolebinding"
+	"github.com/perses/perses/internal/api/shared"
+	databaseModel "github.com/perses/perses/internal/api/shared/database/model"
+	"github.com/perses/perses/internal/api/shared/schemas"
+	"github.com/perses/perses/pkg/model/api"
+	v1 "github.com/perses/perses/pkg/model/api/v1"
+	"github.com/sirupsen/logrus"
+)
+
+type service struct {
+	globalrolebinding.Service
+	dao globalrolebinding.DAO
+	sch schemas.Schemas
+}
+
+func NewService(dao globalrolebinding.DAO, sch schemas.Schemas) globalrolebinding.Service {
+	return &service{
+		dao: dao,
+		sch: sch,
+	}
+}
+
+func (s *service) Create(entity api.Entity) (interface{}, error) {
+	if object, ok := entity.(*v1.GlobalRoleBinding); ok {
+		return s.create(object)
+	}
+	return nil, shared.HandleBadRequestError(fmt.Sprintf("wrong entity format, attempting GlobalroleBinding format, received '%T'", entity))
+}
+
+func (s *service) create(entity *v1.GlobalRoleBinding) (*v1.GlobalRoleBinding, error) {
+	// TODO: validate user + role exists
+	if len(entity.Spec.Subjects) == 0 {
```

Comment:
this check should be at the Unmarshal level I believe. That's because, It's not a functional verification, it's more something that is checking the integrity of the entity itself (and also it is a static check)